### ISSUE TITLE
A fatal exception occurred

### ISSUE THREAD

**Issue by achalpandeyy** [EpicGames/raddebugger#133]
I was going back and forth the "Memory" and "Watch" windows when the exception occurred.

Below is the message copied from the message box.

```
[Window Title]
Fatal Exception

[Content]
A fatal exception (code 0xc0000005) occurred. The process is terminating.

Press Ctrl+C to copy this text to clipboard, then create a new issue in
https://github.com/EpicGames/raddebugger/issues

Call stack:
0. [0x7ff76df21159] df_eval_watch_view_build +21753, df_views.c line 1628
1. [0x7ff76dea4ebc] df_view_ui_Watch +140, df_views.c line 6641
2. [0x7ff76df5e7e6] df_window_update_and_render +149014, df_gfx.c line 5691
3. [0x7ff76dfb70c4] update_and_render +4932, raddbg.c line 285
4. [0x7ff76df69b44] entry_point +5860, raddbg.c line 577
5. [0x7ff76dfb9592] WinMain +930, raddbg_main.cpp line 350
6. [0x7ff76dfc14ee] __scrt_common_main_seh +262, exe_common.inl line 288
7. [0x7ffb307f257d] KERNEL32
8. [0x7ffb3166aa58] ntdll


[Close]
```

**Comment by ryanfleury** [comment]
What version were you using, and/or which commit hash did you build from?

**Comment by mistymntncop** [comment]
Hi, I'm pretty sure I encountered the same bug but on the latest commit.
Unfortunately I just can't figure out how to reproduce it reliably. I was only able to trigger it twice. It happened when I was clicking one of the items in the globals Panel.

I did however have a look at it in remedy and it said that "commit_row" was null on the [same line](https://github.com/EpicGames/raddebugger/blob/c757388bb409e8670b63e31033e0ea0658587484/src/df/gfx/df_views.c#L1669) for the latest commit.

```
[Window Title]
Fatal Exception

[Content]
A fatal exception (code 0xc0000005) occurred. The process is terminating.

Press Ctrl+C to copy this text to clipboard, then create a new issue in
https://github.com/EpicGames/raddebugger/issues

Call stack:
1. [0x7ff6cb71f1ea] df_eval_watch_view_build +33594, df_views.c line 1669
2. [0x7ff6cb69d9cc] df_view_ui_Globals +156, df_views.c line 6766
3. [0x7ff6cb6e2213] df_window_update_and_render +151171, df_gfx.c line 5707
4. [0x7ff6cb722911] update_and_render +4369, raddbg.c line 286
5. [0x7ff6cb739fd0] entry_point +5664, raddbg.c line 577
6. [0x7ff6cb56245d] WinMain +877, raddbg_main.cpp line 348
7. [0x7ff6cb73bc22] __scrt_common_main_seh +262, exe_common.inl line 288
8. [0x7ffd33f47344] KERNEL32
9. [0x7ffd35b026b1] ntdll

Version: 0.9.8 [c757388]

[Close]
```

**Comment by ryanfleury** [comment]
This looks like it's being caused by starting an edit on the value, expression, or view rule of a row in a watch or watch-like tab, scrolling away from it, then clicking somewhere else (thus ending & attempting to commit an edit, for a row which is no longer in view).

There are two options for the debugger here: (a) it can go through with the commit anyways (I'd have to move away from using an actually visualized row's information, since the row which is being edited is out-of-bounds and thus has no UI built for it), or (b) I can abort the edit if the row is out of bounds.

I've gone ahead with (b). While it's somewhat idiosyncratic with the way edits work while a row is *in view* (e.g. where clicking away actually commits the edit), it's a bit more conservative w.r.t. making non-obvious changes to the user, and so I'm leaning in that direction.

So, long story short, this is fixed in e637f7237c338ead02026e15754f44b2cf15429e.

**Comment by achalpandeyy** [comment]
I have had this bug occur again recently
```
[Window Title]
Fatal Exception

[Content]
A fatal exception (code 0xc0000005) occurred. The process is terminating.

Press Ctrl+C to copy this text to clipboard, then create a new issue in
https://github.com/EpicGames/raddebugger/issues

Call stack:
1. [0x7ff683df3eff] df_eval_watch_view_build +22367, df_views.c line 1669
2. [0x7ff683d7540c] df_view_ui_Watch +140, df_views.c line 6721
3. [0x7ff683e323af] df_window_update_and_render +151119, df_gfx.c line 5707
4. [0x7ff683e8bb54] update_and_render +4932, raddbg.c line 285
5. [0x7ff683e3da94] entry_point +5860, raddbg.c line 577
6. [0x7ff683e8e052] WinMain +930, raddbg_main.cpp line 348
7. [0x7ff683e95fae] __scrt_common_main_seh +262, exe_common.inl line 288
8. [0x7ffee01f257d] KERNEL32
9. [0x7ffee0b4aa58] ntdll

Version: 0.9.8 [c757388]

[Close]
```

**Comment by ryanfleury** [comment]
@achalpandeyy You are on a version that was released before this fix was pushed. `git checkout dev`, `git pull`, and `build` to have the fix applied.
